### ISSUE TITLE
proposal: add `Deno.ConfigFile`

### ISSUE THREAD

**Issue by iuioiua** [denoland/deno#24628]
Sometimes projects have to read a `deno.json` file (e.g. in Fresh [here](https://github.com/denoland/fresh/blob/b510113655d0a018eab49c1bb2400bd737eb08a2/tools/release.ts#L28)). In such cases, having a type that makes working with the file easier, from the `Deno.*` namespace, could be quite helpful, and prevent users from having to come up with their own types.

**Comment by dsherret** [comment]
I don't think this should be in the runtime because if breaking changes happen to the config (ex. like lint.files being removed) then the type will change based on what version of Deno the code is run with rather than the version of the deno config that the code supports.

**Comment by iuioiua** [comment]
Isn't that true for any type in the `Deno.*` namespace? Also, to be clear, I'm not strongly in favor of adding this either.

**Comment by dsherret** [comment]
I don't think so because those are runtime APIs. `Deno.ConfigFile` would be a representation of something outside the JS runtime.

**Comment by iuioiua** [comment]
I think it still _might_ provide more value than not having it. @marvinhagemeister, any thoughts?

**Comment by lino-levan** [comment]
I will reiterate my support here, though I understand and sympathize with David's hesitation. As a framework author, it really sucks just having to "guess" where the `deno.json` is going to be, and getting configuration options is only going to be harder as Deno workspaces roll out. I really don't want to end up in a node situation where every framework reinvents node's `package.json` resolving algorithm.

**Comment by BlackAsLight** [comment]
Maybe a new env variable that says the path to the config file that was used? Then when a module needs it, they can read the files content themselves, instead of requiring the user of the module to provide the path/info.

**Comment by iuioiua** [comment]
This is about having an interface that represents the `deno.json` file, not the actual content.

**Comment by lino-levan** [comment]
> This is about having an interface that represents the deno.json file, not the actual content.

Ah sorry, right. I guess it was confusing because you closed #19322 "in favor" of this one when they are not doing the same thing. I misread the initial issue description.
